ground/src/main/java/com/google/android/ground/persistence/remote/firebase/schema/LoiConverter.kt

```diff
@@ -43,10 +44,17 @@ object LoiConverter {
     val geometry = loiProto.geometry.toGeometry()
     val jobId = loiProto.jobId
     val job =
-      DataStoreException.checkNotNull(
-        survey.getJob(jobId),
-        "job $jobId \n loiId = $loiId \n loiProto = $loiProto",
-      )
+      try {
```

Comment:
Why use try / catch instead of just adding `if (job == null)` block here?

But more generally, can't we add this logging higher up in the call stack, in the code I referenced in my original comment? That is where all errors in the conversion process are handled.

Reply:
> Why use try / catch instead of just adding if (job == null) block here?

Because `checkNotNull` internally throws the exception rather than just returning null object. So, this way the job object will never become null


> But more generally, can't we add this logging higher up in the call stack, in the code I referenced in my original comment? That is where all errors in the conversion process are handled.

Updated to the higher entry point